jumpscale/packages/tfgrid_solutions/chats/mattermost.py

```diff
@@ -84,6 +84,7 @@ def domain_select(self):
 
         solution_name = self.solution_name.replace(".", "")
         full_domain = f"{self.threebot_name}-{solution_name}.{self.domain}"
+        self.domain = j.sals.zos.gateway.correct_domain(self.domain)
         while True:
             if j.tools.dnstool.is_free(full_domain):
```

Comment:
shouldn't we validate here the cleanedup domain is free?

Reply:
handled